spam/spam_tutorial.py

```diff
@@ -140,234 +145,406 @@
 # * *Third-party models*: using an pre-trained model (usually a model for a different task than the one at hand)
 # * *Distant supervision*: using external knowledge base
 # * *Crowdworker labels*: treating each crowdworker as a black-box function that assigns labels to subsets of the data
+
+# %% [markdown]
+# ### Recommended practice for LF development
+
+# %% [markdown]
+# Typical LF development cycles include multiple iterations of ideation, refining, evaluation, and debugging.
+# A typical cycle consists of the following steps:
+#
+# 1. Look at examples to generate ideas for LFs
+# 1. Write an initial version of an LF
+# 1. Estimate its performance by looking at examples in the training set or development set
```

Comment:
Huh?  This will be confusing because then will beg the question of why the generative model... I would instead say something like "spot check on a few unlabeled training examples or small labeled dev set if one is available" (important to keep emphasizing that dev set is optional)